parcels/field.py

```diff
@@ -1545,18 +1553,76 @@ def __init__(self, filename, dimensions, indices, netcdf_engine, timestamp=None,
             return self
+
+        # ==== check-opening ==== #
+        try:
+            self.dataset = xr.open_dataset(str(self.filename), decode_cf=True, engine=self.netcdf_engine, chunks={})
+            self.dataset['decoded'] = True
+        except:
+            logger.warning_once("File %s could not be decoded properly by xarray (version %s).\n         It will be opened with no decoding. Filling values might be wrongly parsed."
+                                % (self.filename, xr.__version__))
+            self.dataset = xr.open_dataset(str(self.filename), decode_cf=False, engine=self.netcdf_engine, chunks={})
+            self.dataset['decoded'] = False
+
+        init_chunk_dict = {}
+        if isinstance(self.field_chunksize, dict):
+            init_chunk_dict = self.field_chunksize
+        elif isinstance(self.field_chunksize, tuple):
+            spatStart = 0
+            if 'time' in self.dimensions and self.dimensions['time'] in self.dataset.dims:
+                if(len(self.dimensions) > 2):
+                    init_chunk_dict[self.dimensions['time']] = self.field_chunksize[0]
+                    spatStart += 1
+                else:
+                    init_chunk_dict[self.dimensions['time']] = 1
+            else:
+                raise RuntimeWarning("Did not find {} in NetCDF dims - Please specify field_chunksize as dictionary for NetCDF dimension names.".format(self.dimensions['depth']))
+            if 'depth' in self.dimensions and self.dimensions['depth'] in self.dataset.dims:
+                if(len(self.dimensions) > 3):
+                    init_chunk_dict[self.dimensions['depth']] = self.field_chunksize[1]
+                    spatStart += 1
+                else:
+                    init_chunk_dict[self.dimensions['depth']] = 1
+            else:
+                raise RuntimeWarning("Did not find {} in NetCDF dims - Please specify field_chunksize as dictionary for NetCDF dimension names.".format(self.dimensions['depth']))
+            if 'lat' in self.dimensions and self.dimensions['lat'] in self.dataset.dims:
+                init_chunk_dict[self.dimensions['lat']] = self.field_chunksize[spatStart]
+            else:
+                raise RuntimeWarning("Did not find {} in NetCDF dims. Please specifiy field_chunksize as dictionary for NetCDF dimension names.".format(self.dimensions['lat']))
+            if 'lon' in self.dimensions and self.dimensions['lon'] in self.dataset.dims:
+                init_chunk_dict[self.dimensions['lon']] = self.field_chunksize[spatStart+1]
+            else:
+                raise RuntimeWarning("Did not find {} in NetCDF dims. Please specifiy field_chunksize as dictionary for NetCDF dimension names.".format(self.dimensions['lon']))
+        elif self.field_chunksize == 'auto':
+            av_mem = psutil.virtual_memory().available
+            chunk_cap = av_mem * (1/8) * (1/3)
+            if 'array.chunk-size' in da_conf.config.keys():
```

Comment:
Do we need to throw a warning/info message to notify users that they can improve performance by creating a dask yaml file? Perhaps by adding a link to the website (e.g. http://oceanparcels.org/faq.html#performance; to be created)

Reply:
yes, I can add that.